std/digest/murmurhash.d

```diff
@@ -0,0 +1,751 @@
+/**
+Computes $(LINK2 https://en.wikipedia.org/wiki/MurmurHash, MurmurHash) hashes
+of arbitrary data. MurmurHash is a non-cryptographic hash function suitable
+for general hash-based lookup. It is optimized for x86 but can be used on
+all architectures.
+
+The current version is MurmurHash3, which yields a 32-bit or 128-bit hash value.
+The older MurmurHash 1 and 2 are currently not supported.
+
+MurmurHash3 comes in three flavors, listed in increasing order of throughput:
+$(UL
+$(LI $(D MurmurHash3!32) produces a 32-bit value and is optimized for 32-bit architectures)
+$(LI $(D MurmurHash3!(128, 32)) produces a 128-bit value and is optimized for 32-bit architectures)
+$(LI $(D MurmurHash3!(128, 64)) produces a 128-bit value and is optimized for 64-bit architectures)
+)
```

Comment:
Maybe add something like by default `MurmurHash3!128` picks your current architecture

or group those two categories in the listing, e.g. like this:

- MurmurHash!32 produces a 32-bit value and is optimized for 32-bit architectures
- MurmuHash!128 produces a 128-bit value and selects your current architecture by default
   - MurmurHash!(128, 32) optimized for 32-bit architectures
   - MurmurHash!(128, 64) optimized for 64-bit architectures

We should make it clear that manually passing the second parameter is usually not needed and when one should.